shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java

```diff
@@ -0,0 +1,160 @@
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli;
+
+import alluxio.PropertyKey;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.io.Closer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.Comparator;
+
+/**
+ * A utility to generate property keys to csv files.
+ */
+@ThreadSafe
+public final class ConfigurationDocGenerator {
+  public static final String FILE_HEADER = "propertyName,defaultValue";
+  private static final Logger LOG = LoggerFactory.getLogger(ConfigurationDocGenerator.class);
+
+  private ConfigurationDocGenerator() {
+  } // prevent instantiation
+
+  /**
+   * Writes property key to csv files.
+   *
+   * @param defaultKeys Collection which is from PropertyKey DEFAULT_KEYS_MAP.values()
+   * @param filePath    path for csv files
+   */
+  public static void writeCSVFile(Collection<? extends PropertyKey> defaultKeys, String filePath)
+      throws IOException {
+    if (defaultKeys.size() == 0) {
+      return;
+    }
+
+    FileWriter fileWriter;
+
+    //HashMap for CSV file names
+    Map<String, String> fileNames = new HashMap<>();
+    fileNames.put("user", PathUtils.concatPath(filePath, "user-configuration.csv"));
+    fileNames.put("master", PathUtils.concatPath(filePath, "master-configuration.csv"));
+    fileNames.put("worker", PathUtils.concatPath(filePath, "worker-configuration.csv"));
+    fileNames.put("security", PathUtils.concatPath(filePath, "security-configuration.csv"));
+    fileNames.put("keyvalue", PathUtils.concatPath(filePath, "key-value-configuration.csv"));
+    fileNames.put("common", PathUtils.concatPath(filePath, "common-configuration.csv"));
+
+    //HashMap for FileWriter per each category
+    Map<String, FileWriter> fileWriterMap = new HashMap<>();
+    Closer closer = Closer.create();
+    try {
+      for (Map.Entry<String, String> entry : fileNames.entrySet()) {
+        fileWriter = new FileWriter(entry.getValue());
+        //Write the CSV file header
+        fileWriter.append(FILE_HEADER);
+        //Add a new line separator after the header
+        fileWriter.append("\n");
+        //put fileWriter
+        fileWriterMap.put(entry.getKey(), fileWriter);
+        //register file writer
+        closer.register(fileWriter);
+      }
+
+      //Sort defaultKeys
+      Comparator<PropertyKey> pC = new ConfigurationDocGenerator().new PropertyKeyComparator();
```

Comment:
alluxio.cli.ConfigurationDocGenerator.writeCSVFile(Collection, String) needlessly instantiates a class that only supplies static methods [alluxio.cli.ConfigurationDocGenerator] At ConfigurationDocGenerator.java:[line 83]

I think this should be good
```java
Comparator<PropertyKey> pC = new PropertyKeyComparator();
```